tst/regression/utils/test_case.py

```diff
@@ -108,6 +113,7 @@ def __init__(self,run_test_path,**kwargs):
         self.parameters.mpi_ranks_flag = kwargs.pop('mpirun_ranks_flag')
         self.parameters.num_ranks = int(kwargs.pop('mpirun_ranks_num'))
         self.parameters.mpi_opts = kwargs.pop('mpirun_opts')
+        self.parameters.kokkos_profiling_lib = os.getenv('KOKKOS_PROFILE_LIBRARY')
```

Comment:
So Jonah makes an interesting point above, my thoughts on this are it is probably a good idea to make using the test_case idepotent, seeing as you are changing the environmental variables this could lead to affects difficult to understand behavior. What I would recommend doing is recording the environmental state in the test case and ensuring when your done it looks the same.